### PR TITLE
Compat.Dates

### PR DESCRIPTION
**Pull request by Nosferican**
Added `Compat.Dates` for the main file and the test file.

### CODE REVIEW

**Comment by Nosferican** [comment]
There were test errors on the nightly. Are those related?

**Comment by nalimilan** [comment]
Yes, it seems. Can you avoid doing `using Compat.Dates`, and use `Compat.Dates.X` where needed (with `X` the type or method used)? That way it's clearer where we use Dates (in very few places actually).

File: test/runtests.jl [hunk 1]
```diff
@@ -58,8 +58,8 @@ using Base.Test, Missings
     @test zero(Union{Int, Missing}) === 0
     @test zero(Union{Float64, Missing}) === 0.0
 
-    for T in (subtypes(Dates.DatePeriod)...,
-              subtypes(Dates.TimePeriod)...)
+    for T in (subtypes(Compat.Dates.DatePeriod)...,
```
Comment (nalimilan) [inline test/runtests.jl]: Looks like you need `using Compat` at the top of the file.

Comment (Nosferican) [reply test/runtests.jl]: Thanks.

**Pull request merged**